lib/validate_params/param_builder.rb

```diff
@@ -3,27 +3,57 @@
 module ValidateParams
   module Validatable
     class ParamBuilder
-      def initialize(parent_field: nil, validations:)
-        @parent_field = parent_field
-        @validations = validations
-      end
+      Validation = Struct.new(:field, :type, :options, :children, :parent) do
+        def valid?(value, errors)
+          ParamValidator.call(
+            validation: self,
+            value: value,
+            errors: errors
+          )
 
-      def param(field, type, options = {}, &block)
-        if block
-          yield(ParamBuilder.new(parent_field: field, validations: @validations))
-        else
-          @validations << build_config(field, type, options)
+          if children.any?
+            case type.to_s
+            when "Hash"
+              # Skip in case hash is configured and string is passed
+              unless value.is_a?(String)
```

Comment:
Why we are checking specifically for `String`?

Originally we used `hashlike?(obj)` method which would check for a `Hash` or `ActionController::Parameters`.

Reply:
We have a very strange feature inside this gem where we configure the expected set of parameters for a hash, but instead pass a string value and the system must swallow this feature as an exception and still work as expected.

We even have special tests for this, as you can see, I had to insert such checks in a number of places, to not validate hashes in those places where the passed values are strings.

https://github.com/peopleforce/validate_params/blob/4e2a943e29ab13cb8661f94ab073f537ff998599/spec/fixtures/controllers/with_hash_controller.rb#L13-L16

https://github.com/peopleforce/validate_params/blob/4e2a943e29ab13cb8661f94ab073f537ff998599/spec/validate_params/validatable/default_usage_spec.rb#L11

https://github.com/peopleforce/validate_params/blob/4e2a943e29ab13cb8661f94ab073f537ff998599/spec/validate_params/validatable/default_usage_spec.rb#L87-L97